app/src/main/java/com/orangeelephant/sobriety/ui/screens/create/Create.kt

```diff
@@ -0,0 +1,224 @@
+                                newCounter,
+                                createScreenViewModel.reasonText,
+                                onCounterCreated = { counterId ->
+                                    navController.navigate(route = Screen.CounterFullView.createRoute(counterId = counterId))
+                                })
+                              },
+                    content = {
+                        Text(text = stringResource(id = R.string.create_button))
+                    },
+                    shape = RoundedCornerShape(50.dp),
+                    modifier = Modifier
+                        .size(width = 170.dp, height = 50.dp)
+                )
+            }
+        },
+
+        floatingActionButtonPosition = FabPosition.Center
+
+    ) { innerPadding ->
+
+        Column(
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(top = innerPadding.calculateTopPadding(), bottom = 0.dp)
+        ) {
+
+            // photo
+            Column(
+                verticalArrangement = Arrangement.Top,
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(16.dp)
+            ) {
+                Spacer(modifier = Modifier.height(24.dp))
+                CircleImagePicker(
+                    selectedImageUri = createScreenViewModel.selectedImageUri,
+                    onImageSelected = { uri -> createScreenViewModel.selectedImageUri.value = uri }
+                )
+            }
+
+            // user input
+            Column (
+                verticalArrangement = Arrangement.Bottom,
+                horizontalAlignment = Alignment.CenterHorizontally,
+                modifier = Modifier
+                    .padding(bottom = 100.dp, top = 40.dp, start = 40.dp, end = 40.dp )
+                    .verticalScroll(rememberScrollState()) // Enable vertical scrolling
+
+            ) {
+
+                // name
+                OutlinedTextField(
+                    modifier = Modifier
+                        .fillMaxWidth(),
+                    value = createScreenViewModel.nameText,
+                    onValueChange = { createScreenViewModel.nameText = it },
+                    label = { Text(stringResource(R.string.placeholder_counter_name)) }
+                )
+
+                Spacer(modifier = Modifier.height(8.dp))
+
+                // date
+                ClickableOutlinedTextField(
+                    modifier = Modifier.fillMaxWidth(),
+                    value =  TextFieldValue(convertMillisecondsToDate(createScreenViewModel.dateVal)),
+                    onClick = { isDialogOpen = true },
+                    label = { Text(context.getString(R.string.placeholder_date)) }
+                )
+
+                Spacer(modifier = Modifier.height(8.dp))
+
+                // reasons
+                if (numberOfTextFields > textValues.size) {
```

Comment:
maybe change this to numberOfReasonInputFields or something along those lines since there are other unrelated text fields here

Reply:
yeah fair